google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DeploymentArtifactType.java

```diff
@@ -31,11 +32,13 @@
    * Returns the right {@code DeploymentArtifactType} for the given {@code deployPackage}.
    */
   @NotNull
-  public static DeploymentArtifactType typeForPath(@NotNull File deployPackage) {
-    if (deployPackage.getPath().endsWith(".jar")) {
-      return JAR;
-    } else if (deployPackage.getPath().endsWith(".war")) {
-      return WAR;
+  public static DeploymentArtifactType typeForPath(@Nullable File deployPackage) {
```

Comment:
OK. LGTM

On Wed, Mar 30, 2016 at 2:48 PM, Etan Shaul <notifications@github.com>
wrote:

> In
> google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/DeploymentArtifactType.java
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/565#discussion_r57942295>
> :
>
> > @@ -31,11 +32,13 @@
> >     * Returns the right {@code DeploymentArtifactType} for the given {@code deployPackage}.
> >     */
> >    @NotNull
> > -  public static DeploymentArtifactType typeForPath(@NotNull File deployPackage) {
> > -    if (deployPackage.getPath().endsWith(".jar")) {
> > -      return JAR;
> > -    } else if (deployPackage.getPath().endsWith(".war")) {
> > -      return WAR;
> > +  public static DeploymentArtifactType typeForPath(@Nullable File deployPackage) {
>
> looking at the usages I think it makes sense to let it handle the nulls
> rather than the callers. The callers are already handling
> DeploymentArtifactType.UNKNOWN when needed, so my thinking is to just let
> null just be treated like another unknown type. Let me know if you disagree.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/565/files/b4eb4a51b59de4c0d0374b68fb69ddeb9eda2808#r57942295>
>